core/src/plugins/filed/grpc/CMakeLists.txt

```diff
@@ -0,0 +1,138 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2024-2024 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+include(BareosDisableWarnings)
+include(BareosConfigureFile)
+
+if(HAVE_WIN32)
+  message(STATUS "Disabling grpc plugin as its not supported on windows")
+elseif(HAVE_GRPC)
+  message(STATUS "Using protobuf ${Protobuf_VERSION}")
+  message(STATUS "Using gRPC ${gRPC_VERSION}")
+
+  set(_PROTOBUF_LIBPROTOBUF protobuf::libprotobuf)
+  if(TARGET gRPC::grpc++_reflection)
+    set(_REFLECTION gRPC::grpc++_reflection)
+  else()
+    set(_REFLECTION grpc++_reflection)
+  endif()
```

Comment:
The gRPC targets should always exist. If they don't, then your discovery mechanism is broken (because it skipped creating the proper alias targets).